.github/workflows/golangci.yml

```
@@ -0,0 +1,34 @@
+# SPDX-FileCopyrightText: 2024 Divij Sharma <divijs75@gmail.com>
+# SPDX-License-Identifier: GPL-2.0-only
+
+name: Static Checks and Linting
+
+concurrency:
+  group: lint-${{ github.head_ref }}
+  cancel-in-progress: true
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+  workflow_dispatch:
+
+jobs:
+  golangci-lint:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up Go
+      uses: actions/setup-go@v4
+      with:
+        go-version: '1.20'
+        check-latest: true
+        cache: true
+
+    - name: lint
+      uses: golangci/golangci-lint-action@v3
+      with:
+          version: v1.51
```

```suggestion
          version: v1.55
```
We should use the latest version of a tool for first installation, unless there is a good reason not to.

---

Sure, I was just about to inquire about version updates. I've done some research, and v1.55 looks good to me.